packages/complex-spaces.lua

```diff
@@ -0,0 +1,41 @@
+SILE.nodeMakers["x-spaces-are-nodes"] = pl.class({
+    _base = SILE.nodeMakers.unicode,
+    makeGlue = function (self, item)
+      self:addToken(" ", item)
+      self:makeToken()
+    end
+})
+
+SILE.languageSupport.languages["x-spaces-are-nodes"] = true
+
+SILE.settings.declare({
+  parameter = "shaper.complexspaces",
+  default = true,
+  type = "boolean",
+  help = "Whether the font's space glyph should be emitted, rather than a glue"
+})
+
+if SILE.shaper then
+  local origSpaceNode = SILE.shaper.makeSpaceNode
+
+  SILE.shaper.makeSpaceNode = function (_, options, item)
```

Comment:
Am I understanding this right that it will permanently affect *all* fonts in all languages after load unless the shaper is reloaded? It seems like this should be tied to the current font more than the current language/shaper, no?

Reply:
Yes, this probably needs rethinking to make it per-font. The question is how to automatically turn the feature back off again when a different font is chosen. (And back on again afterwards if we go back to the original font.)

Reply:
Sounds like we need some sort of stack…